src/views/Pathways/Courses/Add/Course.js

```diff
@@ -0,0 +1,69 @@
+import React, {useEffect} from 'react';
+import { useSnackbar } from 'notistack';
+import { withRouter } from 'react-router';
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+import { compose } from 'recompose';
+import { pullAllBy } from 'lodash';
+import { addOrEditCourse } from '../../store';
+import { selectors as layoutSelectors, setRightPaneLoading } from '../../../../layouts/TwoColumn/store';
+import RightPaneWithTitle from '../../../../components/RightPaneWithTitle';
+import FormBuilder from '../../../../components/FormBuilder';
+import history from '../../../../providers/routing/app-history';
+import { getCourseAddFormStructure } from '../form/course';
+import { ngFetch } from '../../../../providers/NGFetch';
+
+
+const CourseAdd = ({ actions, match }) => {
+  const {pathwayId} = match.params;
+  const { enqueueSnackbar } = useSnackbar();
+
+  const [coursesAlreadyPresent, setCoursesAlreadyPresent] = React.useState(null);
+
+  const [submitBtnDisabled, setSubmitBtnDisabled] = React.useState(false);
+  const [allCourses, setAllCourses] = React.useState(null);
+  useEffect(() => {
+    const fetchallCourses = async () => {
+      actions.setRightPaneLoading(true);
+      const responseOfCourses = await ngFetch(`/courses`);
+      setAllCourses(responseOfCourses.availableCourses);
+      const responseOfPathwayCourses = await ngFetch(`/pathways/${pathwayId}/courses`);
+      setCoursesAlreadyPresent(responseOfPathwayCourses);
+      actions.setRightPaneLoading(false);
+    }
+    fetchallCourses();
+  }, [actions,pathwayId]);
+
+
+  pullAllBy(allCourses, coursesAlreadyPresent, 'name')
```

Comment:
All this `pullAllBy` stuff from allCourses on basis of `coursesAlreadyPresent` should be done in the `useEffect` hook itself. Why are you doing it alag se?

Technically you only need to show courses which come as a result of `pullAllBy(allCourses, coursesAlreadyPresent, 'name')` this. Why do you also want to store both allAvailableCourses and alreadyPresentCourses in state? Just store the courses which you need to show in state.